ls1intum/Artemis
# [Bugfix] Category states cannot be changed

### Checklist

### Motivation and Context


Changing the state of a category in the grading page doesn't work anymore.

### Description


- Fix the bug described above
- Fix reset of test cases. The action should be always available not only if changes have been saved **and** a student participation exists
- Change ordering of elements in programming exercise detail view
- Improve spacing in for the option "Enable static code analysis"

### Steps for Testing


1. Log in to Artemis
2. Create a programming exercise with SCA
3. Go to the categories grading page
4. Play around with the options
--> UI should work and categories should update correctly on save
5. Reset the test cases (**Note: Reset for SCA categories is disabled for now**)

### Test Coverage

Nothing changed

### Screenshots


![exercise-creation-sca-unchecked](https://user-images.githubusercontent.com/16407766/100041323-62acb080-2e09-11eb-9d04-e1969591c705.png)
![exercise-detail-view](https://user-images.githubusercontent.com/16407766/100041324-63454700-2e09-11eb-8145-714281ba64f6.png)
